### ISSUE TITLE
sc-mb-import: Nothing happens when clicking import on Safari/Tampermonkey

### ISSUE THREAD

**Issue by CyberSkull** [garylaski/userscripts#14]
Nothing happened when I clicked the import on https://soundcloud.com/selkie_is_a_cyborg/sets/mepple-in-chocotown using Safari 16.6 (18615.3.12.11.2) & Tampermonkey v4.20.6184. I am signed into Soundcloud & MusicBrainz.

**Comment by garylaski** [comment]
I am not having any issues with Tampermonkey on Firefox, so it must be something with Safari, which I am currently unable to test.

**Comment by LexouDuck** [comment]
I am also experiencing the same issue (nothing happens when i click the Import button, not even a debug log in the browser console or anything).

I am on linux though, so instead of Safari have tried it with both Chromium and Firefox (via Tampermonkey) - neither of them worked.
Here is the URL of the album I tried this with: https://soundcloud.com/theofficiallobst/sets/the-journey

**Comment by garylaski** [comment]
I can't seem to replicate this issue. I noticed that with Tampermonkey, I was prompted to give access to the userscript sending requests to musicbrainz.org. This access needs to be granted for the userscript to work. Check to see if the following is set properly (in Tampermonkey -> this userscript -> settings tab):
![2023-09-29T22:23:12,782598667-04:00](https://github.com/garylaski/userscripts/assets/28818063/6a20a419-bd12-4252-bb23-c4c46b3abfff)

**Comment by LexouDuck** [comment]
I have indeed checked that the musicbrainz URL was whitelisted, and it turned out that it already was.

Since you are not able to reproduce the issue, I endeavoured to debug the script myself.
After putting many many `console.log();` lines all over the place, I found where the issue lies.

The album track XHR onload() function silently fails (there is no indication or logging when this happens, i had to add my own try/catch statement to see what was going on). The error comes from line 157 of the script:
```js
let soundcloudTrackData = JSON.parse(response.responseText.split("__sc_hydration =")[1].split(";</script>")[0])[8].data;
```
Here is the (otherwise silent!) error that my newly-added try/catch statement caught:
```
TypeError: Cannot read properties of undefined (reading 'data')
    at Object.onload (userscript.html?name=SoundCloud%253A%2520MusicBrainz%2520import.user.js&id=2e835182-c15b-49a4-a6d5-019e38560ad6:189:147)
    at St (<anonymous>:9:89)
    at <anonymous>:34:72
    at St (<anonymous>:9:89)
    at r (<anonymous>:22:484)
    at <anonymous>:23:112
    at <anonymous>:67:445
    at p (<anonymous>:67:464)
```
You attempt to access element `[8]` of this "hydration" object, but the maximum allowable index is `[7]`:
![image](https://github.com/garylaski/userscripts/assets/35985773/8d1e31bb-659a-456e-85a3-923f3d2f64ad)

This is why it tells us that `...[8].data` is undefined.

I take it what you want to access is the `hydratable:"sound"` object within, right ? It seems pretty fragile to assume that this will always be at index `[8]`, when here we clearly see it is at index `[7]` instead.

So, here is how to fix the issue: simply change the affected fragile code at line 157 to this more robust logic:
```js
let hydrationData = JSON.parse(response.responseText.split("__sc_hydration =")[1].split(";</script>")[0]);
let soundcloudTrackData = hydrationData.find((item) => item.hydratable === "sound").data;
```
I also recommend that you add a try/catch statement inside this XHR `onload()` function, as this will make future debugging much easier.
